.github/workflows/main.yml

```diff
@@ -0,0 +1,232 @@
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ contains(github.event_name, 'pull_request') && github.event.pull_request.head.ref || github.sha }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+#  code-quality:
+#    name: code-quality
+#
+#    runs-on: ubuntu-latest
+#    timeout-minutes: 10
+#
+#    steps:
+#      - name: Check out the repository
+#        uses: actions/checkout@v3
+#        with:
+#          persist-credentials: false
+#
+#      - name: Set up Python
+#        uses: actions/setup-python@v4
+#        with:
+#          python-version: '3.8'
+#
+#      - name: Install python dependencies
+#        run: |
+#          sudo apt-get update
+#          sudo apt-get install libsasl2-dev
+#          python -m pip install --user --upgrade pip
+#          python -m pip --version
+#          python -m pip install pre-commit
+#          pre-commit --version
+#          python -m pip install mypy==0.942
+#          python -m pip install types-requests
+#          mypy --version
+#          python -m pip install -r dev-requirements.txt
+#          dbt --version
+#
+#      - name: Run pre-commit hooks
+#        run: pre-commit run --all-files --show-diff-on-failure
+
+  unit:
+    name: unit test / python ${{ matrix.python-version }}
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+
+    env:
+      TOXENV: "unit"
+      PYTEST_ADDOPTS: "-v --color=yes --csv unit_results.csv"
+
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install python dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install libsasl2-dev
+          python -m pip install --user --upgrade pip
+          python -m pip --version
+          python -m pip install tox
+          tox --version
+
+      - name: Generate session name
+        id: session
+        run: |
+          repo="${GITHUB_REPOSITORY#${GITHUB_REPOSITORY_OWNER}/}"
+          echo "name=${repo}-${GITHUB_RUN_ID}-${GITHUB_RUN_ATTEMPT}" >> "${GITHUB_OUTPUT}"
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-session-name: ${{ steps.session.outputs.name }}
+          role-to-assume: arn:aws:iam::421396179189:role/dbt-glue
+          aws-region: us-east-1
+
+      - name: Run tox
+        run: |
+          export DBT_AWS_ACCOUNT=421396179189
+          export DBT_S3_LOCATION=s3://dbt-glue-ci/data/${GITHUB_RUN_ID}-${GITHUB_RUN_ATTEMPT}/${{ matrix.python-version }}/
+          export DBT_GLUE_ROLE_ARN=arn:aws:iam::421396179189:role/GlueServiceRole
```

Comment:
same as https://github.com/aws-samples/dbt-glue/pull/258/files#r1386403378, put it into secrets.

Reply:
Thanks for the feedback. Addressed.